extensions-core/stats/src/main/java/io/druid/query/aggregation/teststats/ZtestPostAggregator.java

```diff
@@ -0,0 +1,199 @@
+    dependentFields.addAll(sample1Size.getDependentFields());
+    dependentFields.addAll(successCount2.getDependentFields());
+    dependentFields.addAll(sample2Size.getDependentFields());
+
+    return dependentFields;
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return ArithmeticPostAggregator.DEFAULT_COMPARATOR;
+  }
+
+  @Override
+  public Object compute(Map<String, Object> combinedAggregators)
+  {
+    return zScoreTwoSamples(
+        ((Number) successCount1.compute(combinedAggregators)).doubleValue(),
+        ((Number) sample1Size.compute(combinedAggregators)).doubleValue(),
+        ((Number) successCount2.compute(combinedAggregators)).doubleValue(),
+        ((Number) sample2Size.compute(combinedAggregators)).doubleValue()
+    );
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @Override
+  public ZtestPostAggregator decorate(Map<String, AggregatorFactory> aggregators)
+  {
+    return new ZtestPostAggregator(
+        name,
+        Iterables
+            .getOnlyElement(Queries.decoratePostAggregators(Collections.singletonList(successCount1), aggregators)),
+        Iterables.getOnlyElement(Queries.decoratePostAggregators(Collections.singletonList(sample1Size), aggregators)),
+        Iterables
+            .getOnlyElement(Queries.decoratePostAggregators(Collections.singletonList(successCount2), aggregators)),
+        Iterables.getOnlyElement(Queries.decoratePostAggregators(Collections.singletonList(sample2Size), aggregators))
+    );
+  }
+
+  /**
+   * 1. calculating zscore for two-sample Z test. IOW, using z-test statistic
+   * for testing the difference of two population proportions. 2. converting
+   * binary variables (e.g. success or not) to continuous variables (e.g.
+   * conversion rate).
+   *
+   * @param s1count - success count of population 1
+   * @param p1count - sample size of population 1
+   * @param s2count - the success count of population 2
+   * @param p2count - sample size of population 2
+   */
+  private double zScoreTwoSamples(double s1count, double p1count, double s2count, double p2count)
+  {
+    double convertRate1;
+    double convertRate2;
+    Preconditions.checkState(s1count >= 0, "success count can't be negative.");
+    Preconditions.checkState(s2count >= 0, "success count can't be negative.");
+    Preconditions.checkState(p1count >= s1count, "sample size can't be smaller than the success count.");
+    Preconditions.checkState(p2count >= s2count, "sample size can't be smaller than the success count.");
+
+    try {
+      convertRate1 = s1count / p1count;
+      convertRate2 = s2count / p2count;
+
+      return (convertRate1 - convertRate2) /
+             Math.sqrt((convertRate1 * (1 - convertRate1) / p1count) +
+                       (convertRate2 * (1 - convertRate2) / p2count));
+    }
+    catch (IllegalArgumentException ex) {
+      return 0;
+    }
+  }
+
+  @Override
+  public int hashCode()
```

Comment:
Please implement equals() method either. This will remove the potential problem when working with collections.

Reply:
okay